Makefile

```diff
@@ -0,0 +1,50 @@
+# Set Shell to bash, otherwise some targets fail with dash/zsh etc.
+SHELL := /bin/bash
+
+# Disable built-in rules
+MAKEFLAGS += --no-builtin-rules
+MAKEFLAGS += --no-builtin-variables
+.SUFFIXES:
+.SECONDARY:
+.DEFAULT_GOAL := help
+
+include Makefile.vars.mk
+
+.PHONY: help
+help: ## Show this help
+	@grep -E -h '\s##\s' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-20s\033[0m %s\n", $$1, $$2}'
+
+.PHONY: build
+build: build.bin build.docker ## All-in-one build
+
+.PHONY: build.bin
+build.bin: export CGO_ENABLED = 0
+build.bin: fmt vet ## Build binary
+	@go build -o $(BIN_FILENAME) main.go
+
+.PHONY: build.docker
+build.docker: build.bin ## Build docker image
+	docker build -t $(CONTAINER_IMG) .
+
+.PHONY: test
+test: test.go ## All-in-one test
+
+.PHONY: test.go
+test.go: ## Run unit tests against code
```

Comment:
Very nitpicky so feel free to ignore this,  but I was a little confused by the `test.go` target because it looks so much like a file. I would personally prefer a `test-go` (and `build-bin` etc.) just to make it clear that we are not using file targets. This is also generally more in line with [GNU's naming convention](https://www.gnu.org/software/make/manual/html_node/Standard-Targets.html#Standard-Targets)(although there does not seem to be an actual style guide so :shrug:)

Reply:
So the initial motivation comes from a syntax that I kinda like which I''ve seen in Laravel. Hierarchical targets like
```
build
build:bin
build:docker
...
```
However, make doesn't like the `:` as a separator, unless it's escaped with `\:`, but this causes issues like broken shell autocomplete when doing `make <tab>` and also the Makefile becomes way more ugly.
So I thought about just the single dot `.`, but I see the argument that it looks like a filename.
ofc dash would work also, but I see it more like a word separator, less as a hierarchy separator. But I guess that's the `make` way, so I'll change to dash as a separator unless you have other ideas :)